### ISSUE TITLE
F-Droid reproducible build failed

### ISSUE THREAD

**Issue by linsui** [geopaparazzi/smash#270]
https://monitor.f-droid.org/builds/log/eu.hydrologis.smash/72#site-footer Was your apk built from the same commit?

**Comment by moovida** [comment]
Ugh, I was quite sure that I followed the appropriate workflow, but if it failed, there is still something that is wrong.
I wonder if there has been some flutter pub get in between that the IDE did and I didn't notice.

What is the best way for solving this? Should I create another tag and rebuild with the current?
I assume forcing to move the current tag could cause issues in the fdroid build system, right?

**Comment by linsui** [comment]
How did you build the apk? Did you apply the same patch? Is there a script?

**Comment by moovida** [comment]
Oh no, that can only be due to the fact that the gps locator lib has changed. :-(

I am using this script, which mimics the same fdroid does:
https://github.com/geopaparazzi/smash/blob/master/fdroid_build.sh

**Comment by linsui** [comment]
Could you use glob for background_locator_2 version in the script?

**Comment by moovida** [comment]
I am not sure what you mean by that. What is glob?

**Comment by linsui** [comment]
Our script is `cd .pub-cache/git/background_locator*/android` and yours is `cd .pub-cache/git/background_locator_2-3e0e4dcad802b5122d480c940e391c4b518a4f45/android`.

**Comment by moovida** [comment]
Ugh, I didn't notice that :-( sorry

I also now noticed that there is also the path to the repo, wich is not the same anymore, because it is not developed and we had to create a fixed fork. But this also means I will have to change the build script :-( That will take me a bit more time and I assume that I also will need to create a new release, disabling the 1.9.1.

**Comment by moovida** [comment]
No, sorry, forget the last one, all seems to be ok now and I am building it with the same lib.

**Comment by moovida** [comment]
Ok, I updated the script and generated the new old apk. I substituted the old one here on github. Should that work?

**Comment by linsui** [comment]
It works, thanks!

**Comment by moovida** [comment]
Thank you @linsui , you guys are always so kind. Sorry for the mess!

**Comment by moovida** [comment]
This can't be happening. And I even rewrote the script to generate the version name properly. Grrrr

Thanks again, I have updated the apk. :-(

**Comment by moovida** [comment]
I think I can finally close this. The last build and release has been successful without human interaction and support

**Comment by linsui** [comment]
See https://github.com/maelchiotti/LocalMaterialNotes/pull/273 for example.

**Comment by moovida** [comment]
:-( I had no idea this was added.

Would this be happening if the app was signed by f-droid instead of me? And if yes, would it be even possible to switch to f-droid signing? 

I will look in that PR and try to understand what is going on. Thank you/

**Comment by moovida** [comment]
Thanks for the feedback. I will think about it. The problem is that it is hard to find resources to develop the project already, Since every single release up to now has been a disaster (due to my ignorance, no doubt in that) and brought me to release less often due to fear and uncertainty :-), I think it might be for the best of the project.

That said, if I understand it correctly, that Isar thing would not even be solved. It is something flutter added and we need it to be built from source, right?

**Comment by licaon-kter** [comment]
The link above helps right? You and F-Droid both add the build steps to the pipelines.

What do you mean "something flutter added"? You don't control your deps?

**Comment by moovida** [comment]
Yes, it was added by a cache we added and we need. But I will probably look for a substitute that is a bit more slick. 

The thing I do not understand, is why a different app on fdroid that dependes on smashlibs (and therefore on isar), passes. The only difference is that signing occurs on f-droid side. The isar is not built from source in that case.

**Comment by licaon-kter** [comment]
Which app?

**Comment by moovida** [comment]
This one: https://f-droid.org/packages/org.digiagriapp/

We actually added the cache for them.

**Comment by linsui** [comment]
The buid passes because they don't have reproducible build enabled. I bet the app is broken currently because we removed the isar libs.

**Comment by moovida** [comment]
Ok, thank you for the explanation. This sure is another reason to keep reproducible builds going.

**Comment by linsui** [comment]
Oh, you just removed isar from the lib. Thanks!

**Comment by moovida** [comment]
Hi @linsui, yes, I found that more simple :-) 
I was was expecting the build to catch it in a few days, but if you noticed, does that mean that the build passed? 

I looked here: https://monitor.f-droid.org/builds/build
but could not find it.

Anyways, as usual, thanks to both for the help and patience.

**Comment by linsui** [comment]
https://gitlab.com/fdroid/fdroiddata/-/merge_requests/20951 The reproducible build failed with lots of diff. Is your apk built with the same steps? Is there anything changed?

**Comment by moovida** [comment]
:-( I do not understand. I though I did something wrong for release 1.10.0, so I then made sure that I updated everything. So nothing really changed (I added a small bugfix and languages). 
But also I do not understand the log. It starts with:

```
$ for appid in "${CHECKUPDATES_APPIDS[@]}"; do # collapsed multi-line command
2025-03-26 03:43:00,395 INFO: Processing de.ktran.anno1404warenrechner
2025-03-26 03:44:19,598 WARNING: "de.ktran.anno1404warenrechner" branch not found on origin remote:
```

Not sure where to look to find the actual smash related problem.

**Comment by licaon-kter** [comment]
@moovida start with line 816 https://gitlab.com/fdroid/checkupdates-bot-fdroiddata/-/jobs/9524347751#L816 I guess

**Comment by moovida** [comment]
Thanks @licaon-kter . I see there are some play services related things. So it has to be something that changed in the framework and I will have to understand what. I am currently abroad, so I will look into it as soon as I can. Am at Fossgis 2025 presenting SMASH, sad I could not have a release out.

**Comment by linsui** [comment]
We didn't change our script though. Can you make a clean rebuild?

**Comment by linsui** [comment]
You sciprt is not changed and our script is not changed, either. So I thought it should still be reproducible. At least there shouldn't be so many difference. So I guess there are something left from other build. I have no idea why this happened though.

**Comment by linsui** [comment]
Since we build the apk with the same step, there shouldn't be any difference. So I can only guess it's caused by cache. But I have no idea why there can be cache left.

**Comment by moovida** [comment]
So the only possibility is to wait for a next cycle to see if the cache get's cleaned up? 
I am not sure, since I already tried from 1.10.0 and 1.10.1. Nothing changed.

**Comment by licaon-kter** [comment]
you have an APK we can test against? eg. upload it here renamed as . ZIP  (do not archive)

**Comment by moovida** [comment]
For some reason it doesn't allow me to upload even as zip. But actually I am uploading the one I built for the fdroid check:
https://github.com/geopaparazzi/smash/releases/download/version_1.10.1%2B78_fdroid/smash_1.10.1+78_fdroid.apk

Is this ok?

**Comment by licaon-kter** [comment]
nope: https://gitlab.com/fdroid/fdroiddata/-/jobs/9557005163 same play related things, your APK is filled with them

**Comment by moovida** [comment]
OK, so I need to check what has been added in the affected files.
Still do not understand what it going on though.
Will look into it when I am back home.

**Comment by moovida** [comment]
Ok, it gets stranger by the day. I just noticed that a version 1.10.0 is now out on f-droid and it also represents the highest metadata version, while it should be 1.10.1.

**Comment by linsui** [comment]
That's expected. 1.10.0 is built so it's published.

**Comment by moovida** [comment]
Well, that is the first build that gave me "failed" on the fdroid build app. And the reason for which I made a 1.10.1. 
So I am a bit confused.

**Comment by linsui** [comment]
It failed because some error when cloning the submodule.

**Comment by moovida** [comment]
You mean the 1.10.0? Exactly that one failed because of some submodule cloning issue. I thought it was something in my repo, so I updated everything, added a bugfix and tried to release again. And then this.

So what happened is that the 1.10.0, which failed in one cycle, then went well in a following one?

**Comment by linsui** [comment]
Not sure what was the problem but that can be fixed by a fresh clone. Maybe a force-push in the submodule. 🤷

**Comment by moovida** [comment]
Ok, from your last comment I will go for a new build. I just added some fixes and will try a new 1.10.2 release. Fingers crossed.

**Comment by linsui** [comment]
https://github.com/geopaparazzi/smash_import_export_plugins/tree/2a37b197da1ebb72633e40f424a51d139259330d 404...

**Comment by moovida** [comment]
I think I am the worst f-droid contributor ever... :-( 
A push was missing from the import export lib. 
I just pushed it, will that be enough?

**Comment by linsui** [comment]
Lol, no, you are definitely a good contributor! Retrying...

**Comment by linsui** [comment]
It works, thanks!

**Comment by moovida** [comment]
Thank you @linsui !!!
